packages/menu/src/MenuItem.ts

```diff
@@ -276,6 +330,110 @@ export class MenuItem extends LikeAnchor(Focusable) {
         }
     }
 
+    public closeOverlay?: (leave?: boolean) => Promise<void>;
+
+    protected handleSubMenuClick(): void {
+        this.openOverlay({ immediate: true });
+    }
+
+    protected handlePointerenter(): void {
+        if (this.leaveTimeout) {
+            clearTimeout(this.leaveTimeout);
+            delete this.leaveTimeout;
+        } else {
+            this.openOverlay();
+        }
+    }
+
+    protected leaveTimeout?: ReturnType<typeof setTimeout>;
+
+    protected handlePointerleave(): void {
+        if (this.hasSubMenu && this.open) {
+            this.active = true;
+            this.leaveTimeout = setTimeout(() => {
+                delete this.leaveTimeout;
+                if (this.closeOverlay) this.closeOverlay(true);
+            }, 100);
+        }
+    }
+
+    protected handleSubMenuChange = (): void => {
+        this.dispatchEvent(
+            new Event('click', { bubbles: true, composed: true })
+        );
+    };
+
+    protected handleSubMenuPointerenter = (): void => {
+        if (this.leaveTimeout) {
+            clearTimeout(this.leaveTimeout);
+            delete this.leaveTimeout;
+        }
+    };
+
+    public async openOverlay({
+        immediate,
+    }: { immediate?: boolean } = {}): Promise<void> {
+        if (this.hasSubMenu && !this.open) {
+            this.open = true;
+            this.submenu = (
+                this.shadowRoot.querySelector(
+                    'slot[name="submenu"]'
+                ) as HTMLSlotElement
+            ).assignedElements()[0] as Menu;
+            this.submenu.addEventListener(
+                'pointerenter',
+                this.handleSubMenuPointerenter
+            );
+            this.submenu.addEventListener('change', this.handleSubMenuChange);
+            const popover = document.createElement('sp-popover');
+            popover.append(this.submenu);
+            this.submenu.tabIndex = 0;
+            this.submenu.removeAttribute('slot');
+            const closeOverlay = openOverlay(this, 'click', popover, {
+                placement: 'right-start',
```

Comment:
should this be customized to be "left-start" for dir="rtl"? and using "right-start" as default.